kore/test/Test/Kore/Step/Simplification/Integration.hs

```diff
@@ -239,6 +241,70 @@ test_simplificationIntegration =
                     , substitution = mempty
                     }
         assertEqualWithExplanation "" expect actual
+    , testCase "function application introduces definedness condition" $ do
+        let testSortVariable = SortVariableSort $ SortVariable (testId "s")
+            expect =
+                OrPattern.fromPatterns
+                [ Conditional
+                    { term =
+                        Mock.tdivInt
+                            (mkVar Mock.xInt)
+                            (mkVar Mock.xInt)
+                    , predicate =
+                        makeNotPredicate
+                        $ makeEqualsPredicate
+                           (mkVar Mock.xInt)
+                           (Mock.builtinInt 0)
+                    , substitution = mempty
+                    }
+                ]
+        actual <-
+            evaluateWithAxioms
+                ( axiomPatternsToEvaluators
+                    ( Map.fromList
+                        [ (AxiomIdentifier.Application Mock.fIntId
+                          , [ EqualityRule RulePattern
+                                { left = Mock.fInt (mkVar Mock.xInt)
+                                , right = mkVar Mock.xInt
+                                , requires = makeTruePredicate
+                                , ensures = makeTruePredicate
+                                , attributes = def
+                                }
+                            ]
+                          )
+                        , (AxiomIdentifier.Ceil (AxiomIdentifier.Application Mock.tdivIntId)
+                          , [ EqualityRule RulePattern
+                                { left =
+                                    mkCeil testSortVariable
+                                    $ Mock.tdivInt
+                                        (mkVar Mock.xInt)
+                                        (mkVar Mock.yInt)
+                                , right =
+                                    mkCeil testSortVariable
```

Comment:
Do you still need mkCeil here? It looks like it's being applied to a predicate, so it will be removed on the first line of `makeEvaluateNonBoolCeil` anyway.